src/wallet/wallet.cpp

```diff
@@ -2125,32 +2124,28 @@ void CWallet::ReacceptWalletTransactions(interfaces::Chain::Lock& locked_chain)
     // Try to add wallet transactions to memory pool
     for (const std::pair<const int64_t, CWalletTx*>& item : mapSorted) {
         CWalletTx& wtx = *(item.second);
-        CValidationState state;
-        wtx.AcceptToMemoryPool(locked_chain, state);
+        std::string unused_err_string;
+        wtx.SubmitMemoryPoolAndRelay(unused_err_string, false);
     }
 }
 
-bool CWalletTx::RelayWalletTransaction(interfaces::Chain::Lock& locked_chain)
+bool CWalletTx::SubmitMemoryPoolAndRelay(std::string& err_string, bool relay)
 {
     // Can't relay if wallet is not broadcasting
     if (!pwallet->GetBroadcastTransactions()) return false;
-    // Don't relay coinbase transactions outside blocks
-    if (IsCoinBase()) return false;
     // Don't relay abandoned transactions
     if (isAbandoned()) return false;
-    // Don't relay conflicted or already confirmed transactions
-    if (GetDepthInMainChain(locked_chain) != 0) return false;
```

Comment:
You mean relying on `AccessCoin` in `BroadcastTransaction`. Isn't `CCoinsViewCache` a in-memory cache? Is the wallet broadcasting that much transactions so it can be a performance penalty ? `GetDepthInMainChain` is also currently locking the chain via `getBlockHeight`

Reply:
`CCoinsViewCache` hits disk if the in memory cache doesn't contain the coin we're looking for. I doubt the performance matters for the wallet, unlike e.g. peer to peer mempool relay.